packages/core/src/amazonq/extApi.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import vscode from 'vscode'
+import { VSCODE_EXTENSION_ID } from '../shared/utilities'
+import { GenerateAssistantResponseCommandOutput, GenerateAssistantResponseRequest } from '@amzn/codewhisperer-streaming'
+import { FeatureAuthState } from '../codewhisperer/util/authUtil'
+import { ToolkitError } from '../shared'
+
+export interface api {
```

Comment:
Nice work to add types. Do we need to keep this interface 1:1 with `packages/amazonq/src/api.ts`? If so we should make a comment here and there for future devs.

Reply:
`packages/amazonq/src/api.ts` consumes the interface so if we update the interface and don't update `packages/amazonq/src/api.ts` the build will fail. I can still add a comment though